### PR TITLE
docs: next js docs

### PR DESCRIPTION
**Pull request by tobySolutions**
## Why?

This PR adds in Next.js docs PR to the repository for the upcoming release.

## How?

- Add Next.js frameworks docs to the platform next.js docs
- Add the `fleek-next` CLI docs.

Screenshot 2024-10-28 at 23 59 28

Screenshot 2024-10-29 at 00 00 25

## Tickets?

- [Update Next.js docs with new features](https://linear.app/fleekxyz/issue/MKTG-66/update-nextjs-docs-with-new-features)

## Contribution checklist?

- [x] The commit messages are detailed
- [x] The `build` command runs locally
- [x] Assets or static content are linked and stored in the project
- [x] Document filename is named after the slug
- [x] You've reviewed spelling using a grammar checker
- [x] For documentation, guides or references, you've tested the commands and steps
- [x] You've done enough research before writing

## Security checklist?

- [x] Sensitive data has been identified and is being protected properly
- [x] Injection has been prevented (parameterized queries, no eval or system calls)
- [x] The Components are escaping output (to prevent XSS)

### CODE REVIEW

File: src/content/docs/CLI/Fleek-Next-CLI/index.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+1. Before you can deploy your Next.js application you will first need to create a Fleek function with the below command:
+
+   ```bash
+   fleek functions create
+   ```
+
+2. You then copy the name of the Fleek Function you just created because you will use it in the next step. Deploying your Next.js application after all the above steps is as simple as one command:
+
+   ```sh
+   fleek functions deploy --noBundle --name '<function_name>' --path .fleek/dist/index.js
```
Comment (gabrielmpinto) [inline src/content/docs/CLI/Fleek-Next-CLI/index.md:79]: this will be `bundle=true` with the latest CLI version

Comment (tobySolutions) [reply src/content/docs/CLI/Fleek-Next-CLI/index.md:79]: Ok, awesome! Thank you.

File: src/components/FrameworkComparisonTable/constants.ts [hunk 2]
```diff
@@ -4,8 +4,8 @@ export const frameworks: Framework[] = [
   {
     name: 'Next.js',
     ssg: true,
-    ssr: false,
```
Comment (gabrielmpinto) [inline src/components/FrameworkComparisonTable/constants.ts:7]: This made me look into the table, and saw some strange things. Wasn't aware this existed. Do we support SSR for Astro, SvelteKit, Nuxt and Sapper?

Also ISR afaik doesn't apply to most of these frameworks, I think it's just the name of a Next.js feature.

File: src/content/docs/CLI/Fleek-Next-CLI/index.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+desc: An overview of the Fleek Next CLI, a tool that simplifies the deployment of Next.js applications to Fleek.
+---
+
+# Fleek Next adapter
+
+:::info
+Install the [Fleek CLI](/docs) and [login](/docs/cli/#login) locally on your machine to be able to deploy your application.
+:::
+
+The Fleek Next.js adapter allows you to deploy your server-side Next.js application on Fleek. It lets users host fullstack Next.js apps directly from the terminal with an efficiency and edge-compatibility. The adapter helps deploy your Next.js app as a [Fleek Function](/docs/cli/functions) and this in turn offers up the edge runtime built on top the [Fleek Network](/docs/infrastructure/) that powers your Next.js applications.
```
Comment (gabrielmpinto) [inline src/content/docs/CLI/Fleek-Next-CLI/index.md:14]: `and this in turn offers up the edge runtime built on top` I don't get the meaning of this

Comment (gabrielmpinto) [inline src/content/docs/CLI/Fleek-Next-CLI/index.md:14]:
```suggestion
The Fleek Next.js adapter allows you to deploy your server-side Next.js application on Fleek. It lets users host fullstack Next.js apps directly from the terminal with efficiency and edge-compatibility. The adapter helps deploy your Next.js app as a [Fleek Function](/docs/cli/functions) and this in turn offers up the edge runtime built on top the [Fleek Network](/docs/infrastructure/) that powers your Next.js applications.
```

File: src/content/docs/CLI/Fleek-Next-CLI/index.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+date: 2023-01-10
+desc: An overview of the Fleek Next CLI, a tool that simplifies the deployment of Next.js applications to Fleek.
+---
+
+# Fleek Next Adapter
+
+The Fleek Next.js adapter allows you to deploy your server-side Next.js application on Fleek. It lets users host Next.js apps (fullstack and static) directly from the terminal with an efficiency and edge-compatibility. The adapter helps deploy your Next.js app as a [Fleek Function](/docs/cli/functions) and this in turn offers up the edge runtime built on top the [Fleek Network](/docs/infrastructure/) that powers your Next.js applications.
+
+We are going to walk you through the adapter, features it offers, and how to deploy your Next.js app with it on Fleek. This adapter supports both JavaScript and TypeScript Next applications.
+
```
[the remark this answers is not in the document]
Comment (gabrielmpinto) [reply src/content/docs/CLI/Fleek-Next-CLI/index.md:17]: 18.18.2 seems too specific, why not just say `>18`?

File: src/content/docs/CLI/Fleek-Next-CLI/index.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+# Global installation
+npm install @fleek-platform/next -g
+```
+
+## CLI usage
+
+To prepare your Next.js application for deployment on Fleek, follow these steps:
+
+1.  **Configure edge runtime**:
+    For routes with server-side code, add the following code to ensure they run on the edge:
```
Comment (gabrielmpinto) [inline src/content/docs/CLI/Fleek-Next-CLI/index.md:39]:
```suggestion
    For routes with server-side code, add the following code to ensure they are configured to be optimised for edge runtimes:
```

File: src/content/docs/CLI/Fleek-Next-CLI/index.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+```bash
+Transforming code: [████████████████████████████████████████] 100% | ETA: 0s | 100/100
+
+Uploading code to IPFS: [████████████████████████████████████████] 100% | ETA: 0s | 41449/41449
+
+✅ Success! The deployment has been successfully created.
+
+> You can call this Fleek Function by making a request to the following URL
+🔗 https://broad-jackal-colossal.functions.stg.on-fleek-test.app
```
Comment (gabrielmpinto) [inline src/content/docs/CLI/Fleek-Next-CLI/index.md:119]: Please remember to update this with a prod link before merging

File: src/content/docs/CLI/Fleek-Next-CLI/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tobySolutions) [reply src/content/docs/CLI/Fleek-Next-CLI/index.md:17]: Got it, thanks!

File: src/content/docs/Platform/Frameworks/index.mdx [hunk 7]
```diff
@@ -19,7 +19,36 @@ Fleek supports a variety of frameworks and libraries for building sites and appl
...
+
+### Next.js on Fleek
+
+Next.js is a fullstack React framework for the web designed to build feature-rich and scalable, modern frontend applications with ease. Fleek supports Next.js and offers a wide variety of features that make it easy to deploy all your Next.js applications.
+
+#### Getting started
+
+Deploying a Next.js project on Fleek is simple:
+
+1. Get the [`fleek-next-cli`](/docs/cli/fleek-next-cli)
```
Comment (gabrielmpinto) [inline src/content/docs/Platform/Frameworks/index.mdx:32]:
```suggestion
1. Get [`fleek-next`](/docs/cli/fleek-next-cli)
```
It's not a CLI

or
```suggestion
1. Get the [`fleek-next`](/docs/cli/fleek-next-cli) adapter
```

File: src/content/docs/Platform/Frameworks/index.mdx [hunk 8]
```diff
@@ -19,7 +19,36 @@ Fleek supports a variety of frameworks and libraries for building sites and appl
...
+
+#### Features of Next.js on Fleek
+
+:::info
+Incremental Static Regeneration (ISR), Partial Prerendering and Streaming support are being worked on and will be available soon.
+:::
+
+- **Server-Side Rendering (SSR)**: Fleek enables SSR with edge runtimes, letting you render pages dynamically for use cases like authentication and location-based content. SSR scales automatically.
+- **Middleware**: Fleek's Edge Middleware allows custom routing and personalization, making static content adaptable to user preferences or locations.
+- **Service integrations**: Fleek provides integrations with popular services like MongoDB, Sanity, and more, streamlining environment setup and configuration. We offer [Edge SGX](/docs/cli/edge-sgx) for data security and privacy that can be paired with these external services.
```
Comment (gabrielmpinto) [inline src/content/docs/Platform/Frameworks/index.mdx:45]: We don't provide these integrations. I get what you're saying but the wording implies it's something we offer ourselves, like a built-in product.

Maybe replace `provides` with `supports`.

Comment (tobySolutions) [reply src/content/docs/Platform/Frameworks/index.mdx:45]: Got it! Thank you 🙌

File: src/components/FrameworkComparisonTable/constants.ts (hunk 2 above)
Comment (tobySolutions) [reply src/components/FrameworkComparisonTable/constants.ts:7]: Any framework that implements the build output API supports it from what I looked up about it.

Comment (tobySolutions) [reply src/components/FrameworkComparisonTable/constants.ts:7]: Those other frameworks support ISR, but, the ones that don't I added the "-" to.

**Review by KanishkKhurana — APPROVED** [review]

**Review by gabrielmpinto — APPROVED** [review]

**Pull request merged**